doc/guide/javascript.rst

```diff
@@ -0,0 +1,234 @@
+                   "?student=" + studentId +
+                           "&" + query);
+
+    . . . 
+
+The runtime handler code is called by the XBlock's JavaScript code to get the XBlock URL. 
+
+For example, in the example `Thumbs XBlock`_ in the `XBlock SDK`_, the
+`thumbs.js`_ file gets the handler from the XBlock runtime:
+
+.. code-block:: JavaScript
+
+    var handlerUrl = runtime.handlerUrl(element, 'vote');
+
+
+XBlock Children
+*********************
+
+The JavaScript runtime also returns the list of child XBlocks to the XBlock.
+
+From the example above, the following part of the runtime generates and returns
+the list of children to the XBlock:
+
+.. code-block:: JavaScript
+
+    . . .
+
+    children: function(block) {
+          return $(block).prop('xblock_children');
+        },
+    . . . 
+
+NEED EXAMPLE OF XBLOCK USING THIS
+
+WHEN XBLOCK USES THIS VS. MAP
+
+XBlock Child Map
+*********************
+
+The JavaScript runtime also returns the a map of child XBlocks to the running
+XBlock.
+
+From the example above, the following part of the runtime generates and returns
+the list of children to the XBlock:
+
+.. code-block:: JavaScript
+
+    . . .
+
+    childMap: function(block, childName) {
+      var children = this.children(block);
+      for (var i = 0; i < children.length; i++) {
+        var child = children[i];
+        if (child.name == childName) {
+          return child
+        }
+      }
+    }
+    . . . 
+
+NEED EXAMPLE OF XBLOCK USING THIS
+
+
+*********************************
+Using JavaScript in Your XBlock
+*********************************
+
+You provide user interaction in your XBlock through JavaScript.
+
+For example, the `thumbs.js`_ file in the `XBlock SDK`_ provides users with the
+ability to vote up or down on content:
+
+.. code-block:: JavaScript
+  
+  function ThumbsBlock(runtime, element, init_args) {
+    function updateVotes(votes) {
+        $('.upvote .count', element).text(votes.up);
+        $('.downvote .count', element).text(votes.down);
+    }
+
+    var handlerUrl = runtime.handlerUrl(element, 'vote');
+
+    $('.upvote', element).click(function(eventObject) {
+        $.ajax({
+            type: "POST",
+            url: handlerUrl,
+            data: JSON.stringify({voteType: 'up'}),
+            success: updateVotes
+        });
+    });
+
+    $('.downvote', element).click(function(eventObject) {
+        $.ajax({
+            type: "POST",
+            url: handlerUrl,
+            data: JSON.stringify({voteType: 'down'}),
+            success: updateVotes
+        });
+    });
+
+    return {};
+
+You must add the JavaScript file to the fragment in your XBlock.  For example,
+`thumbs.py`_ loads the JavaScript above into the fragment for the student view
+of the Thumbs XBlock:
+
+.. code-block:: Python
+  
+  js_str = pkg_resources.resource_string(__name__, "static/js/src/thumbs.js")
+  frag.add_javascript(unicode(js_str))
+  frag.initialize_js('ThumbsBlock')
```

Comment:
You should probably call out the separate actions of `add_javascript` (which gets the javascript onto the page), and `initialize_js`, which specifies which initialization function to call to initialize the XBlock (this initialization function should be provided by the javascript added in `add_javascript`).